src/main/java/org/opensearch/security/filter/SecurityFilter.java

```diff
@@ -345,6 +345,7 @@ private <Request extends ActionRequest, Response extends ActionResponse> void ap
                         log.info("Transport auth in passive mode and no user found. Injecting default user");
                         user = User.DEFAULT_TRANSPORT_USER;
                         threadContext.putTransient(ConfigConstants.OPENDISTRO_SECURITY_USER, user);
+                        threadContext.putPersistent(ConfigConstants.OPENDISTRO_SECURITY_USER, user);
```

Comment:
I noticed a few of these putPersistent being added. I do not know the full context, but is there any security implications with putting it in persistently? For example maybe stashing the context wouldn't work if the user is persisted into the threadcontext. Any ideas on this usage @cwperks ?

Reply:
I want to add the same code in https://github.com/opensearch-project/security/pull/4896. I added the notion of persistent TC headers in the core a while ago with the intent that these headers are not stashable and I'd like to move to the authenticated user as a non-stashable header.

Its added here because Security is adding an IndexOperationListener on the resourceIndex and the ThreadContext has already been stashed by the sample plugin so that it can write to the resourceIndex. By adding this call here, @DarshitChanpura is guaranteed that he can read the userinfo from the TC inside the IndexOperationListener. 

I believe the `postIndex` hook is running within the same block as the IndexRequest to the resourceIndex.

Reply:
+1 to @cwperks answer. User names are particularly required when creating or listing resources. This would ensure that a user who is current context-user is always present and thus not result in null value. At some point in future, the putTransient calls will be removed completely.